google/cloud/security/iam/inventory/storage.py

```diff
@@ -0,0 +1,662 @@
+    IAM_POLICY = 'iam_policy'
+    GCS_POLICY = 'gcs_policy'
+
+
+class InventoryIndex(BASE):
+    """Represents a GCP inventory."""
+
+    __tablename__ = 'inventory_index'
+
+    id = Column(Integer(), primary_key=True, autoincrement=True)
+    start_time = Column(DateTime())
+    complete_time = Column(DateTime())
+    status = Column(Text())
+    schema_version = Column(Integer())
+    progress = Column(Text())
+    counter = Column(Integer())
+    warnings = Column(Text())
+    errors = Column(Text())
+    message = Column(Text())
+
+    @classmethod
+    def _utcnow(cls):
+        """Return current time in utc.
+
+        Returns:
+            object: UTC now time object.
+        """
+
+        return datetime.datetime.utcnow()
+
+    def __repr__(self):
+        """Object string representation.
+
+        Returns:
+            str: String representation of the object.
+        """
+
+        return """<{}(id='{}', version='{}', timestamp='{}')>""".format(
+            self.__class__.__name__,
+            self.id,
+            self.schema_version,
+            self.start_time)
+
+    @classmethod
+    def create(cls):
+        """Create a new inventory index row.
+
+        Returns:
+            object: InventoryIndex row object.
+        """
+
+        return InventoryIndex(
+            start_time=cls._utcnow(),
+            complete_time=datetime.datetime.utcfromtimestamp(0),
+            status=InventoryState.CREATED,
+            schema_version=CURRENT_SCHEMA,
+            counter=0)
+
+    def complete(self, status=InventoryState.SUCCESS):
+        """Mark the inventory as completed with a final status.
+
+        Args:
+            status (str): Final status.
+        """
+
+        self.complete_time = InventoryIndex._utcnow()
+        self.status = status
+
+    def add_warning(self, session, warning):
+        """Add a warning to the inventory.
+
+        Args:
+            session (object): session object to work on.
+            warning (str): Warning message
+        """
+
+        warning_message = '{}\n'.format(warning)
+        if not self.warnings:
+            self.warnings = warning_message
+        else:
+            self.warnings += warning_message
+        session.add(self)
+        session.flush()
+
+    def set_error(self, session, message):
+        """Indicate a broken import.
+
+        Args:
+            session (object): session object to work on.
+            message (str): Error message to set.
+        """
+
+        self.message = message
+        session.add(self)
+        session.flush()
+
+
+class Inventory(BASE):
+    """Resource inventory table."""
+
+    __tablename__ = 'gcp_inventory'
+
+    # Order is used to resemble the order of insert for a given inventory
+    order = Column(Integer, primary_key=True, autoincrement=True)
+    index = Column(Integer)
+    type_class = Column(String(128))
+    key = Column(String(2048))
+    type = Column(String(256))
+    data = Column(Text())
```

Comment:
qq: Can you please help me to understand if there was any consideration between storing string here vs json type?